### PR TITLE
Improve/status check

### PR DESCRIPTION
**Pull request by rueyaa332266**
Add timeout handle #62 

Change assert method because the message is confused.

Separate the method to reduce Code Climate complexity.

### CODE REVIEW

**Comment by rueyaa332266** [comment]
Change rubocop ModuleLength to 130.

**Comment by rueyaa332266** [comment]
> Category	Count
> Complexity	1

I have no idea to fix this complexity in this time.
Please tell me if you have any idea.
Or we can leave it open and fix it later.

File: lib/bucky/test_equipment/verifications/status_checker.rb [hunk 1]
```diff
@@ -178,6 +180,17 @@ def format_href(base_url, href)
...
         end
+
+        def redirect_and_http_status_check(response, http_status_check_args)
+          fqdn = http_status_check_args[:url][%r{^(https?:\/\/([a-zA-Z0-9\-_.]+))}]
+          redirect_url = response['location']
+          # Add fqdn if location doesn't include fqdn
+          redirect_url = fqdn << redirect_url unless redirect_url.include?('http')
+          puts "  #{http_status_check_args[:url]} ... redirect to #{redirect_url} [#{response.code}:RD]"
+          http_status_check_args[:url] = redirect_url
+          http_status_check_args = http_status_check_args
```
Comment (hikimochi) [inline lib/bucky/test_equipment/verifications/status_checker.rb:191]: Is it need line?

File: lib/bucky/test_equipment/verifications/status_checker.rb [hunk 2]
```diff
@@ -34,7 +34,12 @@ def http_status_check(args)
           # If number of requests is over redirect limit
           return { error_message: "\n[Redirect Error] #{url} is redirected more than #{REDIRECT_LIMIT}" } if redirect_count > REDIRECT_LIMIT
 
-          check_result = check_log_and_get_response(url, device, link_check_max_times, url_log)
+          begin
+            check_result = check_log_and_get_response(url, device, link_check_max_times, url_log)
+          rescue Net::ReadTimeout => e
+            puts "  #{url} ... ReadTimeout"
```
Comment (hikimochi) [inline lib/bucky/test_equipment/verifications/status_checker.rb:40]: Is this stdout needed?
or How about including this message into return error message?

File: lib/bucky/test_equipment/verifications/status_checker.rb [hunk 3]
```diff
@@ -178,6 +180,17 @@ def format_href(base_url, href)
           href.insert(0, '/') unless href.match(%r{^/|^#})
           base_url + href
         end
+
+        def redirect_and_http_status_check(response, http_status_check_args)
```
Comment (hikimochi) [inline lib/bucky/test_equipment/verifications/status_checker.rb:184]: I think we need error handling of redirect loop.(we don't have to implement on this PR)

File: lib/bucky/test_equipment/verifications/status_checker.rb (hunk 2 above)
Comment (rueyaa332266) [reply lib/bucky/test_equipment/verifications/status_checker.rb:40]: It shows the message on console just like status 200 show OK, status 400 show NG.
Just follow the implemented exist code style. It's OK for not showing this message.
And it's already showing in the error message. (by line 41.)

File: lib/bucky/test_equipment/verifications/status_checker.rb (hunk 1 above)
Comment (rueyaa332266) [reply lib/bucky/test_equipment/verifications/status_checker.rb:191]: Thanks, forgot to delete this.

File: lib/bucky/test_equipment/verifications/status_checker.rb (hunk 3 above)
Comment (rueyaa332266) [reply lib/bucky/test_equipment/verifications/status_checker.rb:184]: We can think about it in the next improvement.

**Pull request merged**